example/server.js

```diff
@@ -2,22 +2,51 @@
 
 var Hapi = require('hapi');
 var Hoek = require('hoek');
-var tags = require('../lib/index.js');
+var fs = require('fs');
+var pg = require('pg');
 
 var server = new Hapi.Server();
+var tagsData = JSON.parse(fs.readFileSync('./example/tags.json').toString());
+var tags = require('../lib/index.js');
+var tagsPool = new pg.Pool();
+
+
+function init (port, callback) {
+  server.connection({ port: port });
+
+  server.register([{
+    register: tags,
+    options: { tags: tagsData, tagsPool: tagsPool }
+  }], function (err) {
+    if (err) {
+      return callback(err);
+    }
+
+    server.route([{
+      method: 'GET',
+      path: '/',
+      handler: function (request, reply) {
+        tagsPool.connect(function (connErr, client, done) {
+          Hoek.assert(!connErr, connErr);
+          client.query('select * from tags', function (dberr, res) {
+            Hoek.assert(!dberr, dberr);
+            done();
 
-server.connection({ port: process.env.PORT || 3000 });
+            return reply(res.rows);
+          });
+        });
+      }
+    }, {
+      method: 'GET',
+      path: '/hello',
+      handler: function (request, reply) {
+        reply('hello');
+      }
+    }]);
 
-server.register(tags, function (err) {
-  Hoek.assert(!err, 'error registering plugin');
-});
+    return callback(null, server);
+  });
+}
 
-server.route({
-  method: 'GET',
-  path: '/',
-  handler: function (request, reply) {
-    reply(request.jackmisawesome);
-  }
-});
 
-module.exports = server;
+module.exports = { init: init, tagsPool: tagsPool };
```

Comment:
You can keep the structure you've got (without passing config in the init function) and exporting the tagsPool object but or you can return your tagsPool object in the callback of the init:
```
server.start(function(err){
  return callback(err, server, tagsPool);
})
```